test/common/config/grpc_mux_impl_test.cc

```diff
@@ -527,7 +534,7 @@ TEST_F(GrpcMuxImplTestWithMockTimeSystem, TooManyRequestsWithEmptyRateLimitSetti
 
   // Validate that drain_request_timer is enabled when there are no tokens.
   EXPECT_CALL(*drain_request_timer, enableTimer(std::chrono::milliseconds(100), _));
-  onReceiveMessage(99);
+  onReceiveMessage(100);
```

Comment:
This is happening b/c of the change to bypass queueDiscoveryRequest?

Reply:
I did a bit of digging into why this change. It turns out there are some messages sent when watches get destroyed. Previously we basically had a mix of queue/sendDiscoveryRequest, and rate limiting only applied to some. I've done some refactoring to make this easier (I hope) to understand and added comments, PTAL.